File: src/main/java/br/com/dusty/dkits/listener/gameplay/InventoryClickListener.java
package br.com.dusty.dkits.listener.gameplay;

import br.com.dusty.dkits.gamer.Gamer;
import br.com.dusty.dkits.util.inventory.InventoryUtils;
import br.com.dusty.dkits.util.inventory.WarpMenu;
import br.com.dusty.dkits.warp.Warp;
import br.com.dusty.dkits.warp.Warps;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

public class InventoryClickListener implements Listener {
	
	private static final String TITLE_CONTAINER_INVENTORY = "container.inventory";
	
	@EventHandler
	public void onInventoryClick(InventoryClickEvent event) {
		Player player = (Player) event.getWhoClicked();
		
		Gamer gamer = Gamer.of(player);
		
		if(gamer.getKit().isDummy())
			event.setCancelled(true);
		
		Inventory inventory = event.getClickedInventory();
		
		ItemStack itemStack = event.getCurrentItem();
		
		if(!inventory.getTitle().equals(TITLE_CONTAINER_INVENTORY)){
			
			//Warp menu - main
			if(inventory.getTitle().equals(WarpMenu.TITLE_MAIN)){
				if(itemStack.equals(WarpMenu.BUTTON_GAME))
					player.openInventory(WarpMenu.menuWarpGame(player));
				else if(itemStack.equals(WarpMenu.BUTTON_EVENT))
					player.openInventory(WarpMenu.menuWarpEvent(player));
				
				//Warp menu - game/event
			}else if(inventory.getTitle().equals(WarpMenu.TITLE_GAME) || inventory.getTitle().equals(WarpMenu.TITLE_EVENT)){
				if(itemStack.equals(InventoryUtils.BUTTON_BACK))
					player.openInventory(WarpMenu.menuWarpMain(player));
				else if(!itemStack.equals(InventoryUtils.BACKGROUND))
					for(Warp warp : Warps.WARPS)
						if(warp.getIcon().equals(itemStack)){
							gamer.sendToWarp(warp);
							return;
						}
			}
		}
	}
}

File: src/main/kotlin/br/com/dusty/dkits/listener/gameplay/InventoryClickListener.kt
package br.com.dusty.dkits.listener.gameplay

import br.com.dusty.dkits.gamer.EnumMode
import br.com.dusty.dkits.gamer.Gamer
import br.com.dusty.dkits.util.inventory.InventoryUtils.BUTTON_BACK
import br.com.dusty.dkits.util.inventory.WarpMenu
import br.com.dusty.dkits.warp.Warps
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent

object InventoryClickListener: Listener {

	private val TITLE_CONTAINER_INVENTORY = "container.inventory"

	@EventHandler
	fun onInventoryClick(event: InventoryClickEvent) {
		val player = event.whoClicked as Player

		val gamer = Gamer.of(player)

		if (gamer.kit.isDummy && gamer.mode != EnumMode.ADMIN)
			event.isCancelled = true

		val inventory = event.clickedInventory

		val itemStack = event.currentItem

		if (inventory.title != null && inventory.title != TITLE_CONTAINER_INVENTORY) {

			when (inventory.title) {
				WarpMenu.TITLE_MAIN                       -> {
					when (itemStack) {
						WarpMenu.BUTTON_GAME  -> player.openInventory(WarpMenu.menuWarpGame(player))
						WarpMenu.BUTTON_EVENT -> player.openInventory(WarpMenu.menuWarpEvent(player))
					}
				}
				WarpMenu.TITLE_GAME, WarpMenu.TITLE_EVENT -> {
					when (itemStack) {
						BUTTON_BACK -> player.openInventory(WarpMenu.menuWarpMain(player))
						else        -> Warps.WARPS.filter { it.icon == itemStack }.forEach {
							gamer.sendToWarp(it)
							return
						}
					}
				}
			}
		}
	}
}
